File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from heapq import heapify,heappush,heappop
def solution(participant, completion):
    heapify(participant)
    heapify(completion)
    while completion:
        a = heappop(participant)
        b = heappop(completion)
        if a!=b:
            return a
    return participant[0]

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let map = new Map()
    for(let c of completion){
        if(map.has(c)) map.set(c,map.get(c)+1)
        else map.set(c,1)
    }
    for(let p of participant){
        if(map.get(p)===0 || !map.has(p)) return p
        else map.set(p,map.get(p)-1)
    }
}